### PR TITLE
New ContainerTasker which will use either DockerTasker

### PR DESCRIPTION
**Pull request by rcerven**
Signed-off-by: Robert Cerven <rcerven@redhat.com>

Maintainers will complete the following section:
- [x] Commit messages are descriptive enough
- [x] "Signed-off-by:" line is present in each commit
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [x] Pull request includes link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

**Comment by rcerven** [comment]
This pull request **introduces 1 alert** when merging 0c4a4522daffdb88483e48a326cd5a065a3200f4 into 4d8d213aa4cb56c8150e11496f089b78afc4eee6 - [view on LGTM.com](https://lgtm.com/projects/g/containerbuildsystem/atomic-reactor/rev/pr-01661b94067367fe934375528543c366c1ace821)

**new alerts:**

* 1 for Unused import

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: atomic_reactor/core.py [hunk 1]
```diff
@@ -280,29 +283,131 @@ def export(self, container, chunk_size=1024*2048):
...
+            if self.build_method is None:
+                raise AttributeError("Container task type not yet decided")
+
+            if self.build_method in (CONTAINER_IMAGEBUILDER_BUILD_METHOD,
+                                     CONTAINER_DOCKERPY_BUILD_METHOD):
+                self._tasker = DockerTasker(self.base_url, self.retry_times, self.timeout,
+                                            **self.kwargs)
+                logger.info('ContainerTasker will use DockerTasker')
+            else:
+                raise AttributeError('no tasker')
```
Comment (MartinBasti) [inline atomic_reactor/core.py]: Maybe this should report exact reason, like "No ContainerTasker available for build_method %s"

File: atomic_reactor/core.py [hunk 2]
```diff
@@ -280,29 +283,131 @@ def export(self, container, chunk_size=1024*2048):
...
+    def get_archive(self, *args, **kwargs):
+        return self.tasker.get_archive(*args, **kwargs)
+
+
+class CommonTasker(LastLogger):
+    def __init__(self, **kwargs):
         """
         Constructor
 
         :param base_url: str, docker connection URL
```
Comment (MartinBasti) [inline atomic_reactor/core.py]: Docstring needs update

File: atomic_reactor/core.py [hunk 3]
```diff
@@ -280,29 +283,131 @@ def export(self, container, chunk_size=1024*2048):
...
+            else:
+                raise AttributeError('no tasker')
+
+            info, version = self._tasker.get_info(), self._tasker.get_version()
+            logger.debug(json.dumps(info, indent=2))
+            logger.info(json.dumps(version, indent=2))
+
+        return self._tasker
+
+    def build_image_from_path(self, *args, **kwargs):
```
Comment (MartinBasti) [inline atomic_reactor/core.py]: Can we have more specific param definitions? I'd prefer having common ABC class with abstractmethods that defines common interface for ContainerTasker, DockerTasker and WhateverTasker.

File: atomic_reactor/core.py (hunk 1 above)
Comment (rcerven) [reply atomic_reactor/core.py]: changed

File: atomic_reactor/core.py (hunk 3 above)
Comment (rcerven) [reply atomic_reactor/core.py]: those methods are just basically wrapper methods for methods from specific DockerTasker/PodmanTasker or from CommonTasker
specific param definitions are in there

so when we need to change params for some method, this way we will have to change it just once in specific tasker

File: atomic_reactor/core.py (hunk 2 above)
Comment (rcerven) [reply atomic_reactor/core.py]: updated

File: atomic_reactor/core.py (hunk 3 above)
Comment (twaugh) [reply atomic_reactor/core.py]: It may become clearer exactly what methods and arguments we need in this class once we implement PodmanTasker.

Comment (MartinBasti) [reply atomic_reactor/core.py]: ok

**Comment by MartinBasti** [comment]
Should be occurrences of `DockerTasker` replaced by `ContainerTasker`? Namely:
```
atomic_reactor/buildimage.py:39:        self.tasker = DockerTasker()
atomic_reactor/core.py:81:        self.tasker = DockerTasker()
atomic_reactor/outer.py:44:        self.dt = DockerTasker()
atomic_reactor/outer.py:84:            dt = DockerTasker()
atomic_reactor/plugins/build_orchestrate_build.py:269:        :param tasker: DockerTasker instance  # many occurrences in docstrings
```

**Comment by rcerven** [comment]
@MartinBasti  those are used for standalone building with atomic_reactor
which don't use regular workflow

but I can change comments in plugins about tasker param to be ContainerTasker

**Comment by rcerven** [comment]
changed comments about tasker param in plugins

**Comment by rcerven** [comment]
rn updated: - New ContainerTasker which will use either DockerTasker or PodmanTasker when that is implemented

**Pull request merged**